quickstarts/05-multi-agent-workflow-dapr-workflows/services/client/client.py

```diff
@@ -1,34 +1,68 @@
 #!/usr/bin/env python3
-import requests
-import time
+import json
 import sys
+import time
+import argparse
+from dapr.clients import DaprClient
 
+# Default Pub/Sub component
+PUBSUB_NAME = "messagepubsub"
 
-if __name__ == "__main__":
-    workflow_url = "http://localhost:8004/RunWorkflow"
-    task_payload = {"task": "How to get to Mordor? We all need to help!"}
+def main(orchestrator_topic, max_attempts=10, retry_delay=1):
+    """
+    Publishes a task to a specified Dapr Pub/Sub topic with retries.
+
+    Args:
+        orchestrator_topic (str): The name of the orchestrator topic.
+        max_attempts (int): Maximum number of retry attempts.
+        retry_delay (int): Delay in seconds between attempts.
+    """
+    task_message = {
+        "task": "How to get to Mordor? We all need to help!"
+    }
+
+    time.sleep(5)
 
     attempt = 1
 
-    while attempt <= 10:
+    while attempt <= max_attempts:
         try:
-            print(f"Attempt {attempt}...")
-            response = requests.post(workflow_url, json=task_payload, timeout=5)
-
-            if response.status_code == 202:
-                print("Workflow started successfully!")
-                sys.exit(0)
-            else:
-                print(f"Received status code {response.status_code}: {response.text}")
+            print(f"📢 Attempt {attempt}: Publishing to topic '{orchestrator_topic}'...")
+            
+            with DaprClient() as client:
+                client.publish_event(
```

Comment:
you changed the client app to use pubsub, but it could still work using the rest API as before if the orchestrator agent exposed a service, no? but I guess you made the orchestrator to not expose a service because it uses streaming subscriptions, right? would it be possible to refactor the orchestrator so it still exposes the rest API but uses streaming subscriptions? IMO streaming subscriptions should be always the default regardless of exposing a service or not, because I see the rest service as a convenient API to trigger the agent rather than using pubsub